storage_thumbnail/models/thumbnail_mixin.py

```diff
@@ -54,6 +56,17 @@ def _compute_main_thumbs(self):
                 fname = "thumb_%s_id" % scale
                 rec[fname] = rec._get_thumb(scale_key=scale)
 
+    @api.depends_context("storage_public_url")
+    @api.depends("thumb_medium_id", "thumb_small_id")
+    def _compute_thumb_urls(self):
+        # Use always the internal URL when not required otherwise.
+        url_fname = "internal_url"
+        if self.env.context.get("storage_public_url"):
```

Comment:
PLZ no parameter through context....

IMO internal url is only a good idea when images are stored by odoo. It should therefore  not be the default.

Reply:
> PLZ no parameter through context....

That's the only way to gain control from specific points (eg: related fields). 
Also, I don't see why this should be bad for such a case.

> IMO internal url is only a good idea when images are stored by odoo. It should therefore not be the default.

This is exactly the contrary of what we are experiencing.
CDN traffic burnt for internal ERP usage -> are you sure is a good idea?
CDNs are supposed to be used for large public.

The change here affects only the URLs of the images showed in product views.

Reply:
| The change here affects only the URLs of the images showed in product views.

It"s on purpose it's done as it's. If these images are available on a CDN we don't want to burn odoo's resources to serve theses binaries contents.

Reply:
Now I'm curious about which service do you use because it sounds very cheap (if not free) :smile: 
Anyway, I'll adapt this to become an explicit setting in the backend.

Reply:
You could put your file on a server via ftp and serve theses files via a reverse proxy on a dedicated url, you could use an S3 compatible server as https://min.io/ ...

Reply:
I think you are mixing up "CDN" and "Object Storage".

Reading files publicly from the object storage is _not_ recommended. You need CDN in front of it for public resources.

We do use min.io too (we'll migrate our infra to it actually) and when you add a CDN in front is not for free.
AFAIK every CDN (from AWS or Azure or whatever) is going to cost you based on traffic.

That being said, I'll see if we have alternatives. Having a specific reverse proxy in front of the storage adds another layer to maintain in the stack.

But then, we'll have to enhance the configuration of the backend to be able to configure a CDN URL for public usage and another CDN-like URL for internal usage. ATM you can set only one URL for all resources.

Reply:
@simahawk I thing that something exists to rewrite the url when image is exported for a specific website into shopinvader. https://github.com/shopinvader/odoo-shopinvader/tree/13.0/shopinvader_backend_image_proxy

Reply:
Thanks for the pointer :)

Nevertheless, seems weird to me that the "public URL" of the file is not the one that you configure on the backend.

Also, such feature, the proxy - or the distinction that I make here between public and internal usage URLs - should stay here in the storage repo IMO and computed on a specific field.Then w/ a policy setting on the backend you can decide what to use and when. 

I would keep the changes in this PR and:

1. add a flag `internal_url_for_backend_views` to enable the behavior. When OFF, it behaves as today.
2. add a field `internal_base_url`, default= `/storage.file/$slug`, to give control on internal URLs 

This change would make `s_backend_image_proxy` obsolete because `.url` will always be the public one (normally using a CDN).

What do you think?